internal/compiler/diagnostics.rs

```diff
@@ -285,6 +285,11 @@ impl Diagnostic {
     pub fn source_file(&self) -> Option<&Path> {
         self.span.source_file().map(|sf| sf.path())
     }
+
+    /// return the path of the source file where this error is attached
+    pub fn source_version(&self) -> SourceFileVersion {
```

Comment:
I'm still uncomfortable having this in the public API. Can it somehow be gated? Or what do you think, @tronical

Reply:
Same here. The docs say that this returns a path, but it returns an `Option<i32>` and it's not clear at all what the use of this is.

Reply:
It's gone now.